gitlab/__main__.py

```diff
@@ -1,4 +1,4 @@
-import gitlab.cli
+from . import cli
 
 if __name__ == "__main__":
-    gitlab.cli.main()
+    cli.main()
```

Comment:
Note: repeat of above comment

My convention would lean towards just import the module, not items within the module. As I think it makes the code more readable. We read the code much much more then we ever write it.

My goal (not that I achieve it) is that we make the code is easy to read as possible to reduce "cognitive load". I say this as a `vim` user 😊

Reply:
For me that readability comes with full namespaces (I guess that has its own issues which is why you added relative imports) but not so much with relative imports.

For example with the change below, it's also not clear in the codebase whether we are calling functions from `gitlab.cli` or `gitlab.v4.cli`:

```diff
+   from .v4 import cli

# ...

    parser = _get_base_parser()
-    return gitlab.v4.cli.extend_parser(parser)
+    return cli.extend_parser(parser)
```

Maybe we can poke around in some popular packages and see what they are doing :) I'll have a look, feel free to add more here as well.

Reply:
I checked a few and seems to me importing classes/functions is pretty common? black/requests/click/httpx all seem to almost exclusively do relative imports with functions/classes. mypy is more of a mixed bag and they do both, but not sure how they make the distinction.

Reply:
> For example with the change below, it's also not clear in the codebase whether we are calling functions from `gitlab.cli` or `gitlab.v4.cli`:
> 
> ```diff
> +   from .v4 import cli
> 
> # ...
> 
>     parser = _get_base_parser()
> -    return gitlab.v4.cli.extend_parser(parser)
> +    return cli.extend_parser(parser)
> ```

Well I disagree not being clear (in this case) because they are within three lines of each other 😊 Unfortunately it is not valid syntax to do `from . import v4.cli`.  To enhance clarity could do `from .v4 import cli as v4_cli` ??

But I personally find it much easier to comprehend `cli.main()` vs `main()` when reading the code. As it makes me think, "oh we are probably calling main() in a different module". Where with just `main()` I think, "we are calling the main() function in this module"